src/ensembl/src/content/app/entity-viewer/EntityViewer.tsx

```diff
@@ -91,11 +86,13 @@ const EntityViewer = (props: Props) => {
         genomeId: activeGenomeId,
         entityId: entityIdForUrl
       });
-      props.replace(replacementUrl);
+      dispatch(replace(replacementUrl));
     }
-    props.setDataFromUrl(params);
+    dispatch(setDataFromUrl(params));
   }, [params.genomeId, params.entityId]);
 
+  const SideBarContent = isSidebarModalOpen ? <EntityViewerSidebarModal /> : <GeneViewSideBar />;
```

Comment:
I wonder whether the sidebar modal doesn't more properly belong in the `GeneViewSideBar` component. Given that pressing the download button on the sidebar toolbar will show a very gene-specific download component, I think it tips the balance in favour of having the modal as a child of `GeneViewSideBar`.

Reply:
PS: after some further deliberation, we decided to keep the `EntityViewerSidebarModal` component here rather than in   `GeneViewSideBar`. If you have any strong feelings one way or the other, @imransl @msmanoj , we can be easily persuaded otherwise.